sdk/src/main/java/com/mapbox/maps/StyleObserver.kt

```diff
@@ -76,12 +83,23 @@ internal class StyleObserver(
     loadStyleErrorListener?.onMapLoadError(eventData)
   }
 
+  override fun onStyleDataLoaded(eventData: StyleDataLoadedEventData) {
+    // transition options must be applied after style but before sprite and sources to take effect
```

Comment:
any tip on how it works? does it mean the transitions set after the style is fully loaded does not work any more?

Reply:
how it works now? it does not work now, that's the point of the fix 😄 
technically it could work even now e.g. we do set duration to 0 now (default duration = 300 ms) - style gets loaded pretty fast, let's say in 50 ms - and after it's loaded setting duration to 0 while it's already loading as part of 300 ms - this _seems_ to work now, from what I see.

but in general it's not working and that's why I did attach visuals - if we set duration > 300 ms - it will not take effect before the fix. does it make sense @pengdev?

Reply:
I mean does the `setStyleTransition` change the default transition for all the layer properties when the property is changed after the style is fully loaded?
I think the use case you suggested here is that the transition will be applied while loading a new style for the first time.